pandas/core/missing.py

```diff
@@ -964,3 +974,107 @@ def _rolling_window(a: np.ndarray, window: int):
     shape = a.shape[:-1] + (a.shape[-1] - window + 1, window)
     strides = a.strides + (a.strides[-1],)
     return np.lib.stride_tricks.as_strided(a, shape=shape, strides=strides)
+
+
+def _can_hold_element(values, element: Any) -> bool:
+    """
+    Expanded version of core.dtypes.cast.can_hold_element
+    """
+    from pandas.core.arrays import (
+        ExtensionArray,
+        FloatingArray,
+        IntegerArray,
+    )
+
+    if isinstance(values, ExtensionArray):
+        if hasattr(values, "_validate_setitem_value"):
```

Comment:
i think this is going to be behavior-changing for Categorical (a change that I would like to see eventually)

Reply:
(no longer relevant since now using `can_hold_element`)